bees/mastodonbee/events.go

```diff
@@ -0,0 +1,245 @@
+/*
+ *    Copyright (C) 2020 Christian Muehlhaeuser
+ *                  2020 Nicolas Martin
+ *
+ *    This program is free software: you can redistribute it and/or modify
+ *    it under the terms of the GNU Affero General Public License as published
+ *    by the Free Software Foundation, either version 3 of the License, or
+ *    (at your option) any later version.
+ *
+ *    This program is distributed in the hope that it will be useful,
+ *    but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *    GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *    Authors:
+ *      Christian Muehlhaeuser <muesli@gmail.com>
+ *      Nicolas Martin <penguwin@penguwin.eu>
+ */
+package mastodonbee
+
+import (
+	"context"
+
+	mastodon "github.com/mattn/go-mastodon"
+	"github.com/muesli/beehive/bees"
+)
+
+// handleUpdateEvent handles incoming Toot updates from mastodon from yourself
+// and from people you follow.
+func (mod *mastodonBee) handleStatus(status *mastodon.Status) {
+	ev := bees.Event{
+		Bee:  mod.Name(),
+		Name: "toot_fetched",
+		Options: []bees.Placeholder{
+			{
+				Name:  "id",
+				Value: string(status.ID),
+				Type:  "string",
+			},
+			{
+				Name:  "text",
+				Value: status.Content,
+				Type:  "string",
+			},
+			{
+				Name:  "user_id",
+				Value: string(status.Account.ID),
+				Type:  "string",
+			},
+			{
+				Name:  "username",
+				Value: status.Account.DisplayName,
+				Type:  "string",
+			},
+			{
+				Name:  "reblogs",
+				Value: status.ReblogsCount,
+				Type:  "int64",
+			},
+			{
+				Name:  "favourites",
+				Value: status.FavouritesCount,
+				Type:  "int64",
+			},
+			{
+				Name:  "url",
+				Value: status.URL,
+				Type:  "string",
+			},
+		},
+	}
+	mod.evchan <- ev
+}
+
+func (mod *mastodonBee) handleNotification(notif *mastodon.Notification) {
+	switch notif.Type {
+
+	case "follow":
+		rel, err := mod.client.GetAccountRelationships(context.Background(), []string{string(notif.Account.ID)})
+		if err != nil {
+			mod.LogErrorf("Failed to fetch account relationship at follow event: %v", err)
+			return
+		}
+
+		if len(rel) > 0 {
```

Comment:
I'd just bail out if `len(rel) == 0` here. Then you don't have to indent the following block either.

Reply:
I'm also not sure if we really need the if-case here at all.
We should always fetch exactly one Relationship Entity here - if an error occurs the above code should already have our back